dev/TabView/TabViewItem.cpp

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "common.h"
+#include "TabView.h"
+#include "TabViewItem.h"
+#include "RuntimeProfiler.h"
+#include "ResourceAccessor.h"
+#include "SharedHelpers.h"
+
+TabViewItem::TabViewItem()
+{
+    __RP_Marker_ClassById(RuntimeProfiler::ProfId_TabViewItem);
+
+    SetDefaultStyleKey(this);
+
+    Loaded({ this, &TabViewItem::OnLoaded });
+}
+
+void TabViewItem::OnApplyTemplate()
+{
+    winrt::IControlProtected controlProtected{ *this };
+
+    m_closeButton.set([this, controlProtected]() {
+        auto closeButton = GetTemplateChildT<winrt::Button>(L"CloseButton", controlProtected);
+        if (closeButton)
+        {
+            m_closeButtonClickRevoker = closeButton.Click(winrt::auto_revoke, { this, &TabViewItem::OnCloseButtonClick });
+        }
+        return closeButton;
+    }());
+
+    if (auto tabView = SharedHelpers::GetAncestorOfType<winrt::TabView>(winrt::VisualTreeHelper::GetParent(*this)))
```

Comment:
>SharedHelpers::GetAncestorOfType<winrt::TabView> [](start = 23, length = 48)

I've seen this pattern before and while I think it's probably fine, I also learned there is ItemsControlFromItemContainer static method on ItemsControl, so we should be able to do:

```suggestion
    if (auto tabView = winrt::ItemsControl::ItemsControlFromItemContainer(*this).as<winrt::TabView>())
````